servers/quarkus-cli/src/main/java/org/projectnessie/quarkus/cli/CheckContent.java

```diff
@@ -0,0 +1,204 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.quarkus.cli;
+
+import com.google.protobuf.ByteString;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+import org.projectnessie.versioned.GetNamedRefsParams;
+import org.projectnessie.versioned.Hash;
+import org.projectnessie.versioned.Key;
+import org.projectnessie.versioned.ReferenceInfo;
+import org.projectnessie.versioned.ReferenceNotFoundException;
+import org.projectnessie.versioned.StoreWorker;
+import org.projectnessie.versioned.persist.adapter.ContentAndState;
+import org.projectnessie.versioned.persist.adapter.KeyFilterPredicate;
+import org.projectnessie.versioned.persist.adapter.KeyListEntry;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "check-content",
+    mixinStandardHelpOptions = true,
+    description = "Check content readability of active keys.")
+public class CheckContent extends BaseCommand {
+
+  @CommandLine.Option(
+      names = {"-s", "--separator"},
+      defaultValue = ": ",
+      description = "Output field separator.")
+  private String separator;
+
+  @CommandLine.Option(
+      names = {"-k", "--key"},
+      description =
+          "Content key element (zero or more). If not set, all current keys will be checked.")
+  private List<String> contentKey;
+
+  @CommandLine.Option(
+      names = {"-c", "--show-content"},
+      description = "Print content details for each valid key.")
+  private boolean showContent;
+
+  @CommandLine.Option(
+      names = {"-B", "--batch"},
+      defaultValue = "25",
+      description = "The max number of keys to load at the same time.")
+  private int batchSize;
+
+  @CommandLine.Option(
+      names = {"-W", "--worker-class"},
+      defaultValue = "org.projectnessie.server.store.TableCommitMetaStoreWorker",
+      description = "The class name of the Nessie Store Worker (internal).")
+  private String workerClass;
+
+  @CommandLine.Option(
+      names = {"-r", "--ref"},
+      description = "Reference name to use (default branch, if not set).")
+  private String ref;
+
+  @CommandLine.Option(
+      names = {"-H", "--hash"},
+      description = "Commit hash to use (defaults to the HEAD of the specified reference).")
+  private String hash;
+
+  @CommandLine.Option(
+      names = {"-v", "--verbose"},
+      description = "Print extra error information to STDERR.")
+  private boolean verbose;
+
+  @CommandLine.Option(
+      names = {"-E", "--error-only"},
+      description = "Print status information only for keys with errors")
+  private boolean errorOnly;
+
+  private final AtomicInteger errorDetected = new AtomicInteger();
```

Comment:
Looks like the number of errors is never used. Maybe `AtomicBoolean`?

Reply:
Added `--summary` for this data.